vmm/src/vm.rs

```diff
@@ -1807,30 +1808,38 @@ impl<'a> Vm<'a> {
         // Tell the vCPUs to stop themselves next time they go through the loop
         self.vcpus_kill_signalled.store(true, Ordering::SeqCst);
 
-        // Signal to the vCPU threads. This will interrupt the KVM_RUN ioctl() allowing
-        // the loop to check the boolean set above.
-        for vcpu in self.vcpus.iter() {
+        // Signal to the spawned threads (vCPUs and console signal handler). For the vCPU threads
+        // this will interrupt the KVM_RUN ioctl() allowing the loop to check the boolean set
+        // above.
+        for thread in self.threads.iter() {
             let signum = validate_signal_num(VCPU_RTSIG_OFFSET, true).unwrap();
             unsafe {
-                use std::os::unix::thread::JoinHandleExt;
-                libc::pthread_kill(vcpu.as_pthread_t(), signum);
+                libc::pthread_kill(thread.as_pthread_t(), signum);
             }
         }
 
-        // Wait for all the vCPU threads to finish
-        for vcpu in self.vcpus.drain(..) {
-            vcpu.join().map_err(|_| Error::ThreadCleanup)?
+        // Wait for all the threads to finish
+        for thread in self.threads.drain(..) {
+            thread.join().map_err(|_| Error::ThreadCleanup)?
         }
 
         Ok(exit_behaviour)
     }
 
-    fn os_signal_handler(signals: Signals, console_input_clone: Arc<vm_virtio::ConsoleInput>) {
+    fn os_signal_handler(
+        signals: Signals,
+        console_input_clone: Arc<vm_virtio::ConsoleInput>,
+        quit_signum: i32,
+    ) {
         for signal in signals.forever() {
             if signal == SIGWINCH {
                 let (col, row) = get_win_size();
                 console_input_clone.update_console_size(col, row);
             }
+
+            if signal == quit_signum {
```

Comment:
Using a `match` would be more idiomatic I believe.

Reply:
You can't do a match on the variable it seems:
```
diff --git a/vmm/src/vm.rs b/vmm/src/vm.rs
index 64dcfde..ba10529 100755
--- a/vmm/src/vm.rs
+++ b/vmm/src/vm.rs
@@ -1832,13 +1832,14 @@ impl<'a> Vm<'a> {
         quit_signum: i32,
     ) {
         for signal in signals.forever() {
-            if signal == SIGWINCH {
-                let (col, row) = get_win_size();
-                console_input_clone.update_console_size(col, row);
-            }
-
-            if signal == quit_signum {
-                break;
+            match signal {
+                SIGWINCH => {
+                    let (col, row) = get_win_size();
+                    console_input_clone.update_console_size(col, row);
+                },
+                quit_signum => {
+                    break;
+                }
             }
         }
     }
```

```
warning: unused variable: `quit_signum`
    --> vmm/src/vm.rs:1841:13
     |
1841 |             quit_signum => {
     |             ^^^^^^^^^^^ help: consider prefixing with an underscore: `_quit_signum`
     |
     = note: #[warn(unused_variables)] on by default

warning: unused variable: `quit_signum`
    --> vmm/src/vm.rs:1832:9
     |
1832 |         quit_signum: i32,
     |         ^^^^^^^^^^^ help: consider prefixing with an underscore: `_quit_signum`

   Compiling cloud-hypervisor v0.1.0 (/home/rob/src/cloud-hypervisor)
    Finished dev [unoptimized + debuginfo] target(s) in 3.83s
```